### PR TITLE
*: fix some code smells

### LINKED ISSUE #4302: Fix some issues when running go-critic
go-critic is a good linter. It can find some code smells which are not detected by our CI now.

### PR DESCRIPTION
**Pull request by rleungx**
### What problem does this PR solve?

Close #4302.

### What is changed and how it works?

Internal clean up.

### Check List

Tests

- Unit test

### Release note

```release-note
None
```

### CODE REVIEW

File: tools/pd-ctl/main.go [hunk 1]
```diff
@@ -57,7 +57,7 @@ func main() {
 			fmt.Println(err)
 			return
 		}
-		input = strings.Split(strings.TrimSpace(string(b[:])), " ")
+		input = strings.Split(strings.TrimSpace(string(b)), " ")
```
Comment (lhy1024) [inline tools/pd-ctl/main.go:60]: need to change it?

**Review by lhy1024 — COMMENTED** [review]
the rest LGTM

File: server/core/hot_region_storage.go [hunk 2]
```diff
@@ -162,7 +162,7 @@ func (h *HotRegionStorage) backgroundDelete() {
 	next := time.Date(now.Year(), now.Month(), now.Day(), defaultDeleteTime, 0, 0, 0, now.Location())
 	d := next.Sub(now)
 	if d < 0 {
-		d = d + 24*time.Hour
+		d += 24 * time.Hour
```
Comment (bufferflies) [inline server/core/hot_region_storage.go:165]: some change fing in  weight_moving_average.go  eg. w.sum += n,should this need to uniform

Comment (rleungx) [reply server/core/hot_region_storage.go:165]: Would you like to leave a comment? I think this PR has already changed it which is mentioned in your example.

File: tools/pd-ctl/main.go (hunk 1 above)
Comment (rleungx) [reply tools/pd-ctl/main.go:60]: Can you explain more?

**Review by lhy1024 — APPROVED** [review]

**Review by bufferflies — APPROVED** [review]
lgtm

**Pull request merged**